components/feature/qr/src/main/java/mozilla/components/feature/qr/CustomOverlay.kt

```diff
@@ -0,0 +1,105 @@
+import android.widget.ImageView
+import android.widget.LinearLayout
+
+/**
+ * A custom overlay [View] that sets the opacity of the background to dim it.
+ * This gives a focus like look to the qr reticle.
+ * @property windowFrame a reference to a [Bitmap], used to create the window frame.
+ * @property activity a reference to an [Activity], used to refer the child views of
+ * calling parent view.
+ * @property ALPHA defines the opacity of the background.
+ * @property RADIUS defines the radius of the rectangle corners.
+ */
+
+class CustomOverlay : LinearLayout {
+    private var windowFrame: Bitmap? = null
+    private lateinit var activity: Activity
+
+    companion object {
+        const val RADIUS = 30.0f
+        const val ALPHA = 170
+    }
+
+    constructor(context: Context) : super(context)
+
+    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
+        activity = context as Activity
+    }
+
+    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr)
+
+    /**
+     * This is called before the canvas is drawn. Initializes the windowFrame only once
+     * @param canvas The canvas to draw upon.
+     */
+    override fun dispatchDraw(canvas: Canvas) {
+        super.dispatchDraw(canvas)
+
+        if (windowFrame == null) {
+            createWindowFrame()
+        }
+        canvas.drawBitmap(windowFrame!!, 0f, 0f, null)
+    }
+
+    override fun isEnabled(): Boolean { return false }
+
+    override fun isClickable(): Boolean { return false }
+
+    /**
+     * This function creates the windows frame around our R.id.qr_reticle ImageView
+     * dynamically, so this will scale for all display sizes. Sets an alpha of 170%
+     * to give a dimmed look for the background
+     */
+    private fun createWindowFrame() {
+        val dm = DisplayMetrics()
+        val imageView = activity.findViewById(R.id.qr_reticle) as ImageView
+        val loc = IntArray(2)
+        imageView.getLocationInWindow(loc)
+        activity.windowManager.defaultDisplay.getMetrics(dm)
+        val topOffset = dm.heightPixels - height
+        val bottom = imageView.bottom
+        val right = imageView.right
+
+        windowFrame = Bitmap.createBitmap(width, height, Bitmap.Config.ARGB_8888)
+        val osCanvas = Canvas(windowFrame!!)
+
+        val outerRectangle = RectF(0f, 0f, width.toFloat(), height.toFloat())
+
+        var paint = Paint(Paint.ANTI_ALIAS_FLAG)
+        paint.color = Color.argb(ALPHA, 0, 0, 0)
+        osCanvas.drawRect(outerRectangle, paint)
+
+        paint.xfermode = PorterDuffXfermode(PorterDuff.Mode.SRC_OUT)
+        val innerRectangle = RectF(
+                loc[0].toFloat(),
+                (loc[1] - topOffset).toFloat(),
+                (width - (width - right)).toFloat(),
+                (height - (height - bottom)).toFloat()
+        )
+        osCanvas.drawRoundRect(innerRectangle, Companion.RADIUS, Companion.RADIUS, paint)
```

Comment:
nit: use `RADIUS` directly
```suggestion
        osCanvas.drawRoundRect(innerRectangle, RADIUS, RADIUS, paint)
```